### PR TITLE
Horovod demo

### PR DESCRIPTION
**Pull request by porteratzo**
PR to integrate a new workspace template called torch_llm_horovod, it uses pt_model  as  the collaborator task runner and launches a horovod cluster in the train_batches and validate steps with launch_horovod, this launches in InHorovodrun.py which creates the LLMTrainer and loads the datasets and trains or validates the model. The taskrunner and dataloader in the collaborator handles saving the data and model state for the LLMTrainer to load and return to the taskrunner.

### CODE REVIEW

**Comment by psfoley** [comment]
@porteratzo could you add a description to this PR?

File: openfl-tutorials/experimental/Federeated_Pytorch_LLM_Horovod.py [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+import openfl.native as fx
+import sys
+
+import openfl.interface.workspace as workspace
+import os
+import subprocess
+
+WORKSPACE_PREFIX = os.path.join(os.path.expanduser("~"), ".local", "workspace")
+
+# Run openfl-tutorials/experimental/setup_env.shsetup_env.sh in your venv to setup horovod dependancies
```
Comment (psfoley) [inline openfl-tutorials/experimental/Federeated_Pytorch_LLM_Horovod.py]: copy paste error

```suggestion
# Run openfl-tutorials/experimental/setup_env.sh in your venv to setup horovod dependancies
```

File: openfl-tutorials/experimental/LLM_Horovod.MD
Comment (psfoley) [inline openfl-tutorials/experimental/LLM_Horovod.MD:1]: This file should also be moved to `openfl-workspace/torch_llm_horovod`

**Review by psfoley — CHANGES_REQUESTED** [review]
This is a great contribution, @porteratzo.  A couple small requested changes, and then I think this is ready to merge.

File: openfl-tutorials/experimental/Federeated_Pytorch_LLM_Horovod.py (hunk 1 above)
Comment (porteratzo) [reply openfl-tutorials/experimental/Federeated_Pytorch_LLM_Horovod.py]: Fixed error and added description to PR

File: openfl-tutorials/experimental/LLM_Horovod.MD
Comment (porteratzo) [reply openfl-tutorials/experimental/LLM_Horovod.MD:1]: Moved file to openfl-workspace/torch_llm_horovod

**Review by psfoley — APPROVED** [review]

**Pull request merged**